asn-compiler/src/parser/asn/structs/types/mod.rs

```diff
@@ -137,3 +139,25 @@ impl Asn1ConstructedType {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub(crate) enum Asn1TagMode {
+    Explicit,
+    Implicit,
+}
+
+#[derive(Debug, Clone)]
+pub(crate) enum Asn1TagClass {
+    Universal,
+    Application,
+    ContextSpecific,
+    Private,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) struct Asn1Tag {
+    pub(crate) class: Asn1TagClass,
+    pub(crate) number: u32,
+    pub(crate) mode: Option<Asn1TagMode>,
```

Comment:
I believe - this can simply be a `bool`. Something like `is_explicit` (default `false`) only set when `explicit` found.